packages/pages/src/common/src/template/internal/getTemplateFilepaths.ts

```diff
@@ -0,0 +1,34 @@
+import glob from "glob";
+import path from "path";
+import { Path } from "../../project/path.js";
+
+/**
+ * Get all the template files in the provided template folder path(s).
+ * If there are two files that share the same name between the two
+ * provided template folder paths, only the file in domain template
+ * folder path is collected.
+ *
+ * @param root  The folder path where all template files live
+ * @param domain The folder path where the template files of a specific domain live
+ * @returns a list of template filepaths
+ */
+export const getTemplateFilepaths = (root: Path, domain?: Path): string[] => {
+  const templatesRootFilepaths: string[] = glob.sync(
```

Comment:
would this function be easier to test if it didn't call glob internally but took `templatesRootFilepaths` and `templatesDomainFilepaths` as params or something? Then we wouldn't have to do write our own mock glob

Reply:
maybe that's a little tricky since domain can be undefined though...?

Reply:
I kinda like that we are testing the glob pattern as well currently, to avoid mistakes with accidentally including subfolder/files or other non-template extensions?